.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
```

Comment:
I am not so knowledgeable about this, so I cannot say whether they are excessive or not. 
Just remember I made it simpler here https://github.com/Giveth/giveconomy-balance-aggregator/blob/main/.github/workflows/main-pipeline.yml
Also I build the typescript project and put only the js in the docker image

Reply:
@aminlatifi This is an updated version of the same action that is building the docker image with the addition of prerequisites for building and the support of building on different platforms. I'm using this as a new standard. What we were using is the very first (Old) version. It does the same thing

Reply:
By 'built' phrase, I meant building typescript project to js, not dockerfile!
I used github action to compile the server program and then used the output (./dist folder) in building the image, as it's copying in Dockerfile.